### PR TITLE
update OpenSSL on Windows to 1.1.1w

### PR DESCRIPTION
**Pull request by kevinushey**
### Intent

Addresses the Windows part of https://github.com/rstudio/rstudio/issues/13683.

### Approach

- Rebuild OpenSSL using existing scripts
- Upload newly-built distributable to S3
- Update our usages of OpenSSL in the appropriate locations

### Automated Tests

N/A

### QA Notes

N/A

### Documentation

N/A

### Checklist

- [x] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [x] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [x] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [x] This PR passes all local unit tests

### CODE REVIEW

File: cmake/modules/FindLibR.cmake [hunk 1]
```diff
@@ -107,20 +107,31 @@ else()
       if(NOT LIBR_HOME)
 
          # read home from the registry
-         get_filename_component(LIBR_HOME
-            "[HKEY_LOCAL_MACHINE\\SOFTWARE\\R-core\\R;InstallPath]"
-            ABSOLUTE CACHE)
+         get_filename_component(LIBR_HOME "[HKEY_LOCAL_MACHINE\\SOFTWARE\\R-core\\R;InstallPath]" ABSOLUTE)
```
Comment (kevinushey) [inline cmake/modules/FindLibR.cmake:110]: We shouldn't cache the variable until we're certain we found the "right" one, so I removed the cache step and pushed it down later.

File: dependencies/windows/install-openssl/install-openssl.R [hunk 2]
```diff
@@ -1,19 +1,34 @@
...
 
 # try to find a perl installation directory
-perl <- head(Filter(file.exists, c("C:/Perl64/bin", "C:/Perl/bin")), n = 1)
+perlCandidates <- c(
+   "C:/Perl64/bin",
+   "C:/Perl/bin",
+   "C:/Strawberry/perl/bin"
+)
+
+perl <- head(Filter(file.exists, perlCandidates))
```
Comment (jgutman) [inline dependencies/windows/install-openssl/install-openssl.R:21]: we don't need the n=1 anymore?

**Review by jgutman — APPROVED** [review]

File: cmake/modules/FindLibR.cmake [hunk 3]
```diff
@@ -107,20 +107,31 @@ else()
...
             message(STATUS "Unable to locate R home (not written to registry)")
          endif()
 
          # make sure path exists
-         if (NOT EXISTS "${LIBR_HOME}")
+         if(NOT EXISTS "${LIBR_HOME}")
             message(STATUS "Path to R found in registry '${LIBR_HOME}' doesn't exist")
+            file(GLOB R_INSTALLATIONS "C:/R/*" LIST_DIRECTORIES TRUE)
+            if(R_INSTALLATIONS)
+               list(GET R_INSTALLATIONS 0 LIBR_HOME)
```
[the remark this answers is not in the document]
Comment (kevinushey) [reply cmake/modules/FindLibR.cmake:122]: Yeah, exactly. We could try to be smarter about which R installation we pick, but this is probably good enough.

File: dependencies/windows/install-openssl/install-openssl.R (hunk 2 above)
Comment (kevinushey) [reply dependencies/windows/install-openssl/install-openssl.R:21]: Oops, we do!

Comment (kevinushey) [reply dependencies/windows/install-openssl/install-openssl.R:21]:
```suggestion
perl <- head(Filter(file.exists, perlCandidates), n = 1)
```

**Pull request merged**